clang/lib/Sema/SemaInit.cpp

```diff
@@ -5576,6 +5576,10 @@ static void TryOrBuildParenListInitialization(
       ExprResult ER;
       ER = IS.Perform(S, SubEntity, SubKind,
                       Arg ? MultiExprArg(Arg) : std::nullopt);
+
+      if (ER.isInvalid())
```

Comment:
@AaronBallman -- are you OK with a crash-on-invalid (the one in first post)? The original example we had crashed on valid code, but reducing a 35MB file with a lot of SFINAE is taking weeks with c-reduce.
Thanks for your review :)

Reply:
If you look at `InitializationSequence::Perform` (this function is 1000 lines long), it first checks `Failed()` then proceed to build a whole bunch of things. There are a couple of places it returned `ExprError`. I think it's possible to still be invalid after `!Failed()`.

Reply:
Ah you are correct, sorry! I see now that the common code pattern *is* to check `isInvalid()` on the returned result.

Reply:
For `ExprResult`, I tend to find `isUsable` to be more clear, and I thought is slightly more strict, as it also catches cases where the ER is unset.  That said, we use `isInvalid` all over the place, assuming that it would have been set to SOMETHING.